lib/ssh-ks/client_app.go

```diff
@@ -0,0 +1,345 @@
+package ssh_ks
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/config"
+	"io/ioutil"
+	"os"
+)
+
+// ClientApp represents one client and holds all necessary structures
+type ClientApp struct {
+	// This points to the client holding this structure
+	This *Client
+	// Config holds all servers and clients
+	Config *Config
+	// Private is our private key
+	Private abstract.Secret
+}
+
+// NewClientApp creates a new private/public key pair and returns
+// a ClientApp with an empty Config. It takes a public ssh-key.
+func NewClientApp(sshPub string) *ClientApp {
+	pair := config.NewKeyPair(network.Suite)
+	return &ClientApp{NewClient(pair.Public, sshPub), nil, pair.Secret}
+}
+
+// ReadClientApp searches for the client-app and creates a new one if it
+// doesn't exist
+func ReadClientApp(f string) (*ClientApp, error) {
+	file := expandHDir(f)
+	ca := NewClientApp("TestClient-" + f)
+	_, err := os.Stat(file)
+	if os.IsNotExist(err) {
+		ca.Config = &Config{}
+		return ca, nil
+	}
+	b, err := ioutil.ReadFile(expandHDir(file))
+	if err != nil {
+		return nil, err
+	} else {
+		_, msg, err := network.UnmarshalRegisteredType(b, network.DefaultConstructors(network.Suite))
+		if err != nil {
+			return nil, err
+		}
+		c, ok := msg.(ClientApp)
+		if !ok {
+			return nil, errors.New("Didn't get a ClientApp structure")
+		}
+		ca = &c
+	}
+	return ca, nil
+}
+
+// Write takes a file and writes the clientApp to that file
+func (ca *ClientApp) Write(file string) error {
+	b, err := network.MarshalRegisteredType(ca)
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(expandHDir(file), b, 0660)
+	return err
+}
+
+// NetworkAddServer adds a new server to the Config
+func (ca *ClientApp) NetworkAddServer(s *Server) error {
+	if ca.Config == nil {
+		return errors.New("No config available yet")
+	}
+	dbg.Lvl3("Servers are", ca.Config.Servers)
+	for _, srv := range ca.Config.Servers {
+		// Add the new server to all servers
+		resp, err := networkSendAnonymous(srv.Entity.Addresses[0],
```

Comment:
yeah. we should have that in lib/network => Send anonymously anything you want ! :D